src/gui/socketapi.cpp

```diff
@@ -286,12 +288,26 @@ void SocketApi::command_RETRIEVE_FILE_STATUS(const QString& argument, QIODevice*
     Folder* syncFolder = FolderMan::instance()->folderForPath( argument );
     if (!syncFolder) {
         // this can happen in offline mode e.g.: nothing to worry about
-        statusString = QLatin1String("NOP");
+        statusString = nopString;
     } else {
         const QString file = QDir::cleanPath(argument).mid(syncFolder->cleanPath().length()+1);
-        SyncFileStatus fileStatus = syncFolder->syncEngine().syncFileStatusTracker().fileStatus(file);
 
-        statusString = fileStatus.toSocketAPIString();
+        // future: Send more specific states for paused, disconnected etc.
+        if( syncFolder->syncPaused() ){
```

Comment:
So i'd write this condition:
 
  if(syncFolder->syncPaused()  || !syncFolder->accountState()->isConnected())

So you simplify a lot this code.